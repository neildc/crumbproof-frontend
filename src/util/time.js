import _ from 'lodash';
import moment from 'moment';

export function minToHandM(minutes) {
  const hours = _.floor(minutes / 60);
  const minutesRemainder = minutes % 60;

  return { hours, minutes: minutesRemainder };
}

export function getTotalTimeStr(instructions) {
  const sumMinutes = _.sum(_.map(instructions, 'time_gap_to_next'));
  const { hours, minutes } = minToHandM(sumMinutes);

  if (!hours && !minutes) {
    return null;
  }

  return (hours > 0 ? `${hours}h ${minutes} mins` : `${minutes} mins`);
}

export function dateToday() {
  return (moment().format('MMM Do YY'));
}
