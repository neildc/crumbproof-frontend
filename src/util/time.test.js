import * as time from "./time";

it("converts minutes < 60 to minutes & hours", () => {
  expect(time.minToHandM(30)).toMatchSnapshot();
});

it("converts minutes > 60 to minutes & hours", () => {
  expect(time.minToHandM(320)).toMatchSnapshot();
});

it("null total time string (no instructions)", () => {
  const instructions = [];
  expect(time.getTotalTimeStr(instructions)).toMatchSnapshot();
});

it("null total time string (1 instruction, no time)", () => {
  const instruction = [{ content: "Mix the all the flour and water", id: "1" }];
  expect(time.getTotalTimeStr(instruction)).toMatchSnapshot();
});

it("generates a total time string (1 instruction, with time)", () => {
  const instruction = [
    {
      content: "Mix the all the flour and water",
      id: "1",
      time_gap_to_next: 30
    }
  ];
  expect(time.getTotalTimeStr(instruction)).toMatchSnapshot();
});

it("generates a total time string (multiple instructions)", () => {
  const instructions = [
    { content: "Mix the all the flour and water", id: "1" },
    {
      content: "Cover and rest (up to 30 minutes)",
      id: "2",
      time_gap_to_next: 20
    },
    { content: "Add salt and yeast", id: "3" },
    { content: "Mix", id: "4" },
    { content: "Mix the all the flour and water", id: "5" },
    { content: "Micro rest", id: "6", time_gap_to_next: 5 },
    { content: "Cut and fold a tiny bit more", id: "6", time_gap_to_next: 1 },
    { content: "Cover tub, rest", id: "7", time_gap_to_next: 240 }
  ];

  expect(time.getTotalTimeStr(instructions)).toMatchSnapshot();
});
