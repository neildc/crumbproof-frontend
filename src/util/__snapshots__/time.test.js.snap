// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converts minutes < 60 to minutes & hours 1`] = `
Object {
  "hours": 0,
  "minutes": 30,
}
`;

exports[`converts minutes > 60 to minutes & hours 1`] = `
Object {
  "hours": 5,
  "minutes": 20,
}
`;

exports[`generates a total time string (1 instruction, with time) 1`] = `"30 mins"`;

exports[`generates a total time string (multiple instructions) 1`] = `"4h 26 mins"`;

exports[`null total time string (1 instruction, no time) 1`] = `null`;

exports[`null total time string (no instructions) 1`] = `null`;
